File: shop_app/lib/entitys/client_entity.dart
class ClientEntity {
  int? id;
  int? idShop;
  String? name;
  String? phone;
  String? address;
  double? oweMoney;
  double? totalPurchased;

  ClientEntity(
      {this.id,
        this.name,
        this.phone,
        this.idShop,
        this.address,
        this.oweMoney,
        this.totalPurchased});

  ClientEntity.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    phone = json['phone'];
    idShop = json['idShop'];
    address = json['address'];
    oweMoney = json['oweMoney'];
    totalPurchased = json['totalPurchased'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['phone'] = this.phone;
    data['idShop'] = this.idShop;
    data['address'] = this.address;
    data['oweMoney'] = this.oweMoney;
    data['totalPurchased'] = this.totalPurchased;
    return data;
  }
}

File: tienda-backend-apirest/src/main/java/com/bolsadeideas/springboot/backend/apirest/models/entity/ClientEntity.java
package com.bolsadeideas.springboot.backend.apirest.models.entity;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "client")
public class ClientEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long idShop;
    private String name;
    private String phone;
    private String address;
    private double oweMoney;
    private double totalPurchased;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public double getOweMoney() {
        return oweMoney;
    }

    public void setOweMoney(double oweMoney) {
        this.oweMoney = oweMoney;
    }

    public double getTotalPurchased() {
        return totalPurchased;
    }

    public void setTotalPurchased(double totalPurchased) {
        this.totalPurchased = totalPurchased;
    }

    public Long getIdShop() {
        return idShop;
    }

    public void setIdShop(Long idShop) {
        this.idShop = idShop;
    }

    @Override
    public String toString() {
        return "ClientEntity{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", phone='" + phone + '\'' +
                ", address='" + address + '\'' +
                ", oweMoney=" + oweMoney +
                ", totalPurchased=" + totalPurchased +
                '}';
    }
}
